api/utilities.py

```diff
@@ -153,24 +154,27 @@ async def add_request_parameters(
     if standard_names:
         request.filter["standard_name"].values.extend(split_and_strip(standard_names))
 
+    if levels:
+        request.filter["level"].values.extend(get_z_levels_or_range(levels))
+
     if methods:
         request.filter["function"].values.extend(split_and_strip(methods))
 
     if periods:
-        request.filter["period"].values.extend(await get_periods_from_request(periods))
+        request.filter["period"].values.extend(get_periods_from_request(periods))
 
 
-async def get_periods_from_request(periods: str | None) -> list[str]:
+def get_periods_from_request(periods: str) -> list[str]:
     split_on_slash = periods.split("/")
     if len(split_on_slash) == 1:
-        return split_and_strip(periods)
+        return [str(iso_8601_duration_to_seconds(period)) for period in split_and_strip(periods)]
     elif len(split_on_slash) == 2:
-        return await get_iso_8601_range(split_on_slash[0].upper(), split_on_slash[1].upper())
+        return get_iso_8601_range(split_on_slash[0].upper(), split_on_slash[1].upper())
```

Comment:
Upper is needed for `isodate` library as an input, otherwise it throws an error. We want to allow requests with lowercase periods, so I moved the upper method to the `iso_8601_duration_to_seconds` function to make it clearer: https://github.com/EURODEO/e-soh/pull/172/commits/1fb617a9345d68e3eb2a2d9674467bac06602d40